testapi.php

```diff
@@ -0,0 +1,117 @@
+        }
+        return $this;
+    }
+    public function getMessage()
+    {
+        return $this->_message;
+    }
+    public function getAllSampleRequests()
+    {
+        if (!$this->_token) {
+            return false;
+        }
+        $headers = array("Authorization: Bearer $this->_token");
+        $searchCriteria = [
+            'searchCriteria' => [
+                'sortOrders' => [
+                    0 => [
+                        'field'     => 'created_at',
+                        'direction' => 'DESC',
+                    ]
+                ],
+                'currentPage' => 1
+            ]
+        ];
+        $searchCriteriaQuery = http_build_query($searchCriteria);
+        $requestUrl = $this->_magentoBaseUrl . $this->_magentoApiController . 'askQuestion/search?'.$searchCriteriaQuery;
+        var_dump($requestUrl);
+        $ch = curl_init($requestUrl);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $result = curl_exec($ch);
+        $result = json_decode($result);
+        return $result->items;
+    }
+    public function getSampleRequest($askQuestionId)
+    {
+        if (!$this->_token) {
+            return false;
+        }
+        $headers = array("Authorization: Bearer $this->_token");
+        $requestUrl = $this->_magentoBaseUrl . $this->_magentoApiController . 'askQuestion/'.$askQuestionId;
+        var_dump($requestUrl);
+        $ch = curl_init($requestUrl);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $result = curl_exec($ch);
+        $result = json_decode($result);
+        return $result;
+    }
+    public function saveSampleRequest($askQuestion)
+    {
+        if (!$this->_token) {
+            return false;
+        }
+        $requestUrl = $this->_magentoBaseUrl . $this->_magentoApiController . 'askQuestion';
+        if ($askQuestion->id) {
+            $requestUrl .= '/'.$askQuestion->id;
+        }
+        var_dump($requestUrl);
+        $data_string = json_encode(
+            ['askQuestion' => $askQuestion]
+        );
+        $headers = array(
+            "Authorization: Bearer $this->_token",
+            'Content-Type: application/json',
+            'Content-Length: ' . strlen($data_string)
+        );
+        $ch = curl_init($requestUrl);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $data_string);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        $response = curl_exec($ch);
+        $response = json_decode($response);
+        return $response;
+    }
+}
+$magento = new MagentoAPI('kalexey', 'kalexey2');
+$sampleRequest = $magento->getSampleRequest(1);
+var_dump($sampleRequest);
+unset($sampleRequest->created_at);
+$sampleRequest->product_name .= ' (CHANGED)';
```

Comment:
There was no field $sampleRequest->product_name in your response. Please check the fields names, specify another field to check if data is saving